### DISCUSSION TITLE
Unresponsive keyboard after inactivity

### DISCUSSION THREAD

**Discussion by Kawsay** [i3/i3#6205] [Q&A]
Hi,

I just installed i3wm through `apt` on a fresh Debian 12 laptop, and sometimes after a brief inactivity period (watching a 10min video for example), my keyboard stops working. I'm using a Thinkpad P50 laptop and the keyboard seems to be a "HP, Inc HyperX Pulsefire Core Keyboard"

I believe it's linked to i3 because when I start a new tty I can type again.

I've only tried `# dpkg-reconfigure udev` which didn't fix the issue and didn't went deeper as I lack of knowledge and couldn't find related resources online.

Here's the version I'm using:
```
$ i3 --moreversion 2>&- || i3 --version
Binary i3 version:  4.22 (2023-01-02) © 2009 Michael Stapelberg and contributors
Running i3 version: 4.22 (2023-01-02) (pid 1135)
Loaded i3 config:
  /home/kawsay/.config/i3/config (main) (last modified: Tue 13 Aug 2024 11:06:20 AM CEST, 96347 seconds ago)

The i3 binary you just called: /usr/bin/i3
The i3 binary you are running: i3
```

Here are the custom configuration I've added to the default one:
```
# Enable touchpad tap to click
exec --no-startup-id xinput set-prop "Synaptics TM3149-002" "libinput Tapping Enabled" 1

# Use Mod + Tab to switch back and forth between last used workspaces
bindsym $mod+Tab workspace back_and_forth

# Use Polybar
exec_always --no-startup-id $HOME/.config/polybar/launch.sh &

# Display wallpapper
exec_always --no-startup-id feh --bg-fill $HOME/Pictures/mountain.jpg

# Display gap between windows if multiple are displayed
gaps inner 2px
# gaps outer 2px # for extra gap around the windows (not between)
smart_gaps on

# Transparency
exec --no-startup-id picom -b

# Remove top title bar
for_window [class="^.*"] border pixel 0

# Prevent screensaver
exec --no-startup-id caffeine

# Activate numeric keypad at startup
exec --no-startup-id xdotool key --clearmodifiers Num_Lock

# Shutdown
bindsym $mod+Shift+s exec systemctl poweroff -i
```

Could you help me debugging and solving this so I can better understand it and have a working keyboard ?

**Comment by Kawsay** [comment]
Issue tends to happen mostly when my computer turns back on after hibernation

**Comment by chip-e** [comment]
Hey, 

I had a similar issue with similar set-up (Debian 12 with i3), except I am on a Thinkpad X230. I managed to fix it by editing `/etc/default/grub` and changing the line beginning with `GRUB_CMDLINE_LINUX_DEFAULT=` 

to `GRUB_CMDLINE_LINUX_DEFAULT="quiet splash i8042.direct i8042.dumbkbd"`

See: https://askubuntu.com/questions/916465/ubuntu-17-04-keyboard-not-responding-after-suspend

**Reply by Kawsay** [reply]
Hey, thanks for your answer ! 

It does not seems to be a definitive answer to this problem. While it fixes it most of the time, it still happens that my keyboard does not respond after hibernation

**Comment by Certainty1396** [comment]
I encountered the same problem that after a while the keyboard cannot input any letters but bindsym key worked well. I have to refresh i3 config file to make it work again.

**Reply by TacoBlayno** [reply]
I'm having the same problem except on Arch Linux using systemd-boot instead of Grub and I turned off auto-sleep and auto-hyrbid in /etc (don't remember which file, but it related to system services). Despite this it still sleeps and after a while i3 input will stop working and I can't get it to show up again, but I can open new ttys, so input isn't broken on the system, only for i3. Restarting service and renstalling i3 and X11 didn't fix it either, but those are the only things having problems. It's obviously a problem with i3 though because everything else works and i3 has always done things that don't conform with the rest of the system.

**Comment by lfk-dev** [comment]
I can confirm similar issue. For me it happens after waking up from sleep.
I still can access different ttys. It's interesting that plugging a keyboard to my laptop fixes the issue, that is both laptop's and external keyboard start working again.
```
$ i3 --moreversion 2>&- || i3 --version
Binary i3 version:  4.24 (2024-11-06) © 2009 Michael Stapelberg and contributors
Running i3 version: 4.24 (2024-11-06) (pid 1294)
Loaded i3 config:
  /home/luk/.config/i3/config (main) (last modified: Fri 31 Oct 2025 10:35:15 PM CET, 79442 seconds ago)

The i3 binary you just called: /usr/bin/i3
The i3 binary you are running: i3
```
